.github/actions/get-ado-token/action.yml

```diff
@@ -0,0 +1,34 @@
+name: Get Azure DevOps Access Token
+inputs:
+  client-id:
+    description: "The client ID of the application calling Azure DevOps"
+    required: true
+  tenant-id:
+    description: "The tenant ID of the application calling Azure DevOps"
+    required: true
+  organization:
+    description: "The Azure DevOps organization to authenticate with"
+    required: true
+outputs:
+  token:
+    description: "The access token to authenticate with Azure DevOps"
+    value: ${{ steps.ADOAuth.outputs.token }}
+
+runs:
+  using: "composite"
+  steps:
+    - name: OIDC Login with AzPowershell
+      uses: azure/login@v2
+      with:
+        client-id: ${{ inputs.client-id }}
+        tenant-id: ${{ inputs.tenant-id }}
+        allow-no-subscriptions: true
+        enable-AzPSSession: true

```

Comment:
just curious, will the `AzPSSession` be closed automatically after usage?

Reply:
Great question! It will indeed be closed after usage. This task injects what's known as a "post" action at the end of the workflow run. This is visible [here](https://github.com/AzureAD/microsoft-authentication-cli/actions/runs/9231515558/job/25401557591?pr=383) under "Post Get Azure DevOps Access Token"

Reply:
great. good to know!